### DISCUSSION TITLE
HPC for Moose error！

### DISCUSSION THREAD

**Discussion by jakson121** [idaholab/moose#28345] [Q&A Getting Started]
### Check these boxes if you have followed the posting rules.

- [x] I have consulted the [Posting Guidelines](https://github.com/idaholab/moose/discussions/18270).
- [X] I have searched the [Discussions Forum](https://github.com/idaholab/moose/discussions) and [MOOSE Framework Troubleshooting](https://mooseframework.inl.gov/help/troubleshooting.html) and have not found what I was looking for
- [X] Q&A Getting Started is the most appropriate category for my question (trouble installing, beginner user, ...)

### Issue or question about MOOSE

I am setting up MOOSE using the HPC method from the official website. I have already completed the setup of libmesh, PETSc, and WASP, but I encountered an error while building MOOSE. Please help me！
![1723286109146](https://github.com/user-attachments/assets/8837a7b1-4d82-40a9-b4dc-af896d533f4d)

**Comment by GiudGiud** [comment]
Hello

Is this a fresh install?
These errors are often seen when leftover compiled objects are present from a previous installation of the code

**Reply by jakson121** [reply]
yes，I am installing MOOSE using the cluster method, but I encountered this issue during compilation. Do you know how to solve it?

**Reply by milljm** [reply]
Can you provide the results of a diagnostic?

```bash
cd ~/projects/moose/scripts
./diagnostics.sh
```

**Reply by jakson121** [reply]
##################################################################################################
Influential Environment Variables

CONDA_CHANNEL=https://conda.software.inl.gov/public
CURL_CA_BUNDLE=
LD_LIBRARY_PATH=/usr/local/lib:/usr/local/lib:
MOOSE_JOBS=6
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/wsl/lib:/mnt/c/Program Files/Common Files/Oracle/Java/javapath:/mnt/d/Software/bin/:/mnt/c/Windows/system32:/mnt/c/Windows:/mnt/c/Windows/System32/Wbem:/mnt/c/Windows/System32/WindowsPowerShell/v1.0/:/mnt/c/Windows/System32/OpenSSH/:/mnt/c/Program Files (x86)/NVIDIA Corporation/PhysX/Common:/mnt/c/Program Files/NVIDIA Corporation/NVIDIA NvDLISR:/mnt/d/Software new/MATLAB/runtime/win64:/mnt/d/Software new/MATLAB/bin:/mnt/d/Software new/Git/cmd:/mnt/c/Program Files/CMake/bin:/mnt/c/Users/AppData/Local/Microsoft/WindowsApps:/mnt/d/Software new/Microsoft VS Code/bin:/mnt/d/Software new/mingw-19.0/MinGW/bin:/snap/bin
PETSC_ARCH=test
PETSC_DIR=/home/junjie/software/petsc-3.21.3
REQUESTS_CA_BUNDLE=
SSL_CERT_FILE=

##################################################################################################
Compiler(s) (CC CXX FC F77 F90):

CC              not set
CXX             not set
FC              not set
F77             not set
F90             not set

FAIL: One or more compiler environment variables not set

##################################################################################################
Python Sanity Checks

Verify `/usr/bin/env python3 --version` (reporting as: Python 3.10.12),
matches versions for: `which python3 && which python`


WARNING: `python` does not exist
This does not mean there will be a failure, but some shebangs in some python
files may still be relying on calling: `/usr/bin/env python` (Python 2.x era)
OK

##################################################################################################
Python Modules (TestHarness, run-ability)

OK

**Reply by milljm** [reply]
This feels like you are running in a WSL session. Which is fine, but usually folks install our Conda packages when doing so. It is not a hard requirement of course, one could install everything from scratch. And if you have, what seems to be missing here is setting the necessary compiler environment variables.

I am not sure what to have you set those to, but we can see what happens:

```bash
export CC=mpicc CXX=mpicxx FC=mpifort F90=mpif90 F77=mpif77
# and then run diags again, see what it detects:
./diagnostics.sh
```
